python/tank/platform/engine.py

```diff
@@ -487,10 +493,46 @@ def execute_in_main_thread(self, func, *args, **kwargs):
         """
         Execute the specified function in the main thread when called from a non-main
         thread.  This will block the calling thread until the function returns.
-        
+
         Note, this currently only works if Qt is available, otherwise it just
         executes on the current thread.
-        
+
+        # ---------------------------------------------------------------------------------
+        # The following test checks that execute_in_main_thread is itself thread-safe!  It
+        # runs a simple test a number of times in multiple threads and ensures the result
+        # returned is as expected.  If this isn't the case it will print out one or more:
+        #     'Result mismatch...'
+        #
+        import threading
+        import random
+        import time
+        import sgtk
+        from sgtk.platform.qt import QtCore
+        
+        NUM_TEST_THREADS=20
+        NUM_THREAD_ITERATIONS=30
+        
+        def run_in_main_thread(v):
+            print "Running", v
+            return v
```

Comment:
nah, it's just an example test and I didn't want it to accidentally go through execute_in_main_thread via the engine.log_info!